### PR TITLE
feat(perf-issue): rerevert and fix post process forwarder handling multiple groups

### PR DESCRIPTION
**Pull request by barkbarkimashark**
Rerevert of https://github.com/getsentry/sentry/pull/39519, but includes a fix which originally caused the issue.

### CODE REVIEW

File: src/sentry/eventstream/kafka/postprocessworker.py [hunk 1]
```diff
@@ -73,6 +74,7 @@ def dispatch_post_process_group_task(
     is_new_group_environment: bool,
     primary_hash: Optional[str],
     skip_consume: bool = False,
+    group_states: Optional[GroupStates] = None,
```
Comment (barkbarkimashark) [inline src/sentry/eventstream/kafka/postprocessworker.py:77]: This param should have been included in the original PR.

The appropriate tests for `postprocessworker` were modified, but the actual function wasn't updated.

**Review by wedamija — APPROVED** [review]
This looks good to me, thanks for the extra tests. Let's try deploy this again tomorrow morning

**Comment by nikhars** [comment]
Can you confirm that you were able to run the post process forwarder locally with your changes and did not see any crashes? I usually run Sentry and pass it some transactions/events to ensure everything works as expected.

**Comment by barkbarkimashark** [comment]
> Can you confirm that you were able to run the post process forwarder locally with your changes and did not see any crashes? I usually run Sentry and pass it some transactions/events to ensure everything works as expected.

Yup, ran this locally as you suggested and threw in some error events to be processed and everything shows up in the UI.

**Pull request merged**